aws/resource_aws_s3_bucket.go

```diff
@@ -728,6 +728,11 @@ func resourceAwsS3BucketRead(d *schema.ResourceData, meta interface{}) error {
 		if err := d.Set("cors_rule", rules); err != nil {
 			return err
 		}
+	} else {
```

Comment:
Nitpick: The logic can be simpler if `rules` was instantiated before the if statement and the `d.Set()` was moved outside the if:

```go
rules := make([]map[string]interface{}, 0)
if cors.CORSRules != nil {
  for _, ruleObject := range cors.CORSRules {
    // ...
  }
  // remove d.Set() here
}
if err := d.Set("cors_rule", rules); err != nil {
  return fmt.Errorf("error setting cors_rule: %s", err)
}
```